content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,563 @@
+
+> Before starting this article, I would like to disclaim that this tutorial is greatly inspired by a online tutorial I 
+created for Poutyne framework. Also, the content is based on a recent article I wrote with Marouane Yassine. However, 
+there are differences between the present work and the two others, I've tried to put more insights for the less technical reader in this one.
+>
+Sequential data, such as addresses, are pieces of information that are deliberately given in a specific order. In other words, they are sequences with a particular structure; and knowing this structure is crucial for predicting the missing entries of a given truncated sequence. For example, 
+when writing an address, we know, in Canada, that after the civic number (e.g. 420), we have the street name (e.g. du Lac).
+Hence, if one was asked to complete an address containing only a number, she could reasonably assume that the next information that should be added to the sequence is a street name. Various modelling approaches have been proposed to make predictions over sequential data. Still, more recently, deep learning models known as Recurrent Neural Network (RNN) has been introduced for this type of data.
+
+Training an RNN requires various tricks (padding and packing) that we will explore in this article. First, let's 
+state our problem, and later on, we will discuss what an actual RNN or LSTM is.
+
+## Address Tagging
+Address tagging is the task of detecting, by tagging, the different parts of an address such as the civic number, 
+the street name or the postal code (or zip code). The following figure shows an example of such a tagging.
+
+![address parsing canada](address_parsing.png)
+
+Since addresses are written in a predetermined sequence, RNN is the best way to crack this problem. But to decode the output of the RNN, we also need another component, a fully-connected layer. 
+So, our architecture, we will be composed of an RNN and a fully-connected layer. 
+
+## RNN
+Speaking of RNN, what is it?
+
+In brief, RNN is neural networks where connections between nodes form a temporal sequence. It means that this type of networks
+allow previous outputs to be used as inputs of the previous information 
+([For more about RNN](https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-recurrent-neural-networks)). 
+
+But, instead of using a vanilla RNN, we use a variant of it, know as a long short-term memory (LSTM) that is known to have
+better stability with gradient update (vanishing and exploding gradient) by using gates 
+([to learn more about LSTM](http://colah.github.io/posts/2015-08-Understanding-LSTMs/)). 
+
+Also, for now, let's simply use a single layer unidirectional LSTM. We will, later on, explore the use of more layers and bidirectional approach. 
```

Comment:
Ce serait bien d'avoir une phrase du genre : LSTMs can be uni- or bidirectional. We'll explain that concept later. For now, let us consider the most simple ones, unidirectional LSTMs. Tho j'ai pas lu la suite donc je suis pas certain que ça fit avec le texte.

Reply:
Je le dit plus tard effectivement.